staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -720,6 +728,10 @@ func (r *crdHandler) getOrCreateServingInfoFor(uid types.UID, name string) (*crd
 	}
 
 	for _, v := range crd.Spec.Versions {
+		if !v.Served {
```

Comment:
should we move this to *just* before we actually construct the `storages[v.Name] = customresource.NewStorage(`?

other possibly impactful things are done in this loop I'm not sure we should skip, like registering equivalent kinds for webhooks, etc... skipping that has visible impact